### PR TITLE
cmd_runner: add __call__ method to invoke context

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Method `context()` renamed to `__call__()`, but retaining an alias from the old name to the new one, for compatibility.
That change will allow a more concise idiom when running the commands.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/module_utils/cmd_runner.py

### CODE REVIEW

**Comment by russoz** [comment]
@felixfontein do you mean that uses `__call__`? The existing code that uses `cmd_runner` were written when `context`.

**Comment by russoz** [comment]
And I have unit tests for both usages.

**Comment by felixfontein** [comment]
I mean changing some of the existing code that uses `context()` so that it uses `__call__()`, yes.

**Comment by russoz** [comment]
> I mean changing some of the existing code that uses `context()` so that it uses `__call__()`, yes.

That will prove that Python works as expected. :-P

Anyways, done.

File: changelogs/fragments/4791-cmd-runner-callable.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - cmd_runner module utils - add __call__ method to invoke context (https://github.com/ansible-collections/community.general/pull/4791).
```
Comment (felixfontein) [inline changelogs/fragments/4791-cmd-runner-callable.yaml:2]:
```suggestion
  - cmd_runner module utils - add ``__call__`` method to invoke context (https://github.com/ansible-collections/community.general/pull/4791).
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for this improvement!

**Pull request merged**